security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/SecurityAnalysisCommandOptions.java

```diff
@@ -0,0 +1,205 @@
+        requireNonNull(outputFile);
+        this.outputFile = i -> outputFile;
+        this.outputFileFormat = requireNonNull(format);
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions taskBasedOnIndex(int taskCount) {
+        return task(i -> new Partition(i + 1, taskCount));
+    }
+
+    public SecurityAnalysisCommandOptions task(Function<Integer, Partition> task) {
+        this.task = requireNonNull(task);
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions task(Partition task) {
+        requireNonNull(task);
+        this.task = i -> task;
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions resultExtension(String extensionName) {
+        this.resultExtensions.add(requireNonNull(extensionName));
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions resultExtensions(Collection<String> extensionNames) {
+        this.resultExtensions.addAll(requireNonNull(extensionNames));
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions violationType(LimitViolationType violationType) {
+        this.violationTypes.add(requireNonNull(violationType));
+        return this;
+    }
+
+    public SecurityAnalysisCommandOptions violationTypes(Collection<LimitViolationType> violationTypes) {
+        this.violationTypes.addAll(requireNonNull(violationTypes));
+        return this;
+    }
+
+    private String pathToString(Path path) {
+        return (absolutePaths ? path.toAbsolutePath() : path).toString();
+    }
+
+    public SimpleCommand toCommand() {
+        Objects.requireNonNull(caseFile, "Case file is not defined.");
+        Preconditions.checkArgument(task == null || taskCount == null,
+                "Options task and task-count may not be defined together.");
+
+        SimpleCommandBuilder commandBuilder = new SimpleCommandBuilder()
+                .id(id)
+                .program(itoolsCommand != null ? itoolsCommand : "itools")
+                .arg("security-analysis")
+                .option(CASE_FILE_OPTION, pathToString(caseFile));
+
+        setOptionIfPresent(commandBuilder, PARAMETERS_FILE_OPTION, parametersFile, this::pathToString);
+        setOptionIfPresent(commandBuilder, CONTINGENCIES_FILE_OPTION, contingenciesFile, this::pathToString);
+        setOptionIfPresent(commandBuilder, OUTPUT_FILE_OPTION, outputFile, this::pathToString);
+        setOptionIfPresent(commandBuilder, OUTPUT_FORMAT_OPTION, outputFileFormat);
+        setOptionIfPresent(commandBuilder, OUTPUT_LOG_OPTION, logFile, this::pathToString);
+        if (!resultExtensions.isEmpty()) {
+            commandBuilder.option(WITH_EXTENSIONS_OPTION, String.join(",", resultExtensions));
+        }
+        if (!violationTypes.isEmpty()) {
+            commandBuilder.option(LIMIT_TYPES_OPTION, violationTypes.stream().map(LimitViolationType::name).collect(Collectors.joining(",")));
+        }
```

Comment:
Maybe you can introduce a setOptionIfPresent that works with containers, to have the same syntax for all parameters?

Reply:
I just tried but it does not make it very readable, because then we need to define a lambda for the serialization to string, as an argument to the function.